lib/providers/profiles_provider.dart

```diff
@@ -149,22 +148,39 @@ class ProfilesProvider with ChangeNotifier {
       if (response.statusCode < 400) {
         var decodedBody = json.decode(response.body);
         List<Profile> fetchedProfiles = [];
+        List<Profile> sortedProfiles = [];
         List<Transaction> fetchedTransactions = [];
 
         if (decodedBody is List) {
-          for (var i = 0, j = 0; i < decodedBody.length; i++, j++) {
-            var element = decodedBody[i];
+          for (var element in decodedBody) {
             var profileGuid = element["guid"];
 
-            var profileTransactions = await _fetchTransactions(profileGuid);
+            fetchedProfiles.add(
+              Profile(
+                guid: profileGuid,
+                name: element["name"],
+                balance: element["balance"],
+                monster: Monsters.blue,
```

Comment:
Instead of doing this you can make it as a default value in the fromJson method or in constructor